pkg/util/k8sutil/tpr_util.go

```diff
@@ -110,8 +99,11 @@ func updateClusterTPRObject(k8s *unversioned.Client, host, ns string, e *spec.Et
 		return nil, err
 	}
 	req.Header.Set("Content-Type", "application/json")
+	return sendReq(k8s.Client, req)
+}
 
-	resp, err := k8s.Client.Do(req)
+func sendReq(httpcli *http.Client, req *http.Request) (*spec.EtcdCluster, error) {
```

Comment:
i am fine with adding a func call readOutCluster(io.Reader) spec.EtcdCluster; and a func called toTPRError that translate error to tpr related error.